File: application/app/src/main/java/com/yahyeet/boardbook/model/entity/Game.java
package com.yahyeet.boardbook.model.entity;

import java.util.ArrayList;
import java.util.List;

public class Game extends AbstractEntity {
	private String name;
	private String description;
	private int minPlayers;
	private int maxPlayers;
	private int difficulty;
	private List<GameTeam> teams = new ArrayList<>();
	private List<Match> matches = new ArrayList<>();

	public Game(String name,
							String description,
							int difficulty,
							int minPlayers,
							int maxPlayers) {
		this.name = name;
		this.description = description;
		this.difficulty = difficulty;
		this.minPlayers = minPlayers;
		this.maxPlayers = maxPlayers;
	}

	public Game(String id) {
		super(id);
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public int getDifficulty() {
		return difficulty;
	}

	public void setDifficulty(int difficulty) {
		this.difficulty = difficulty;
	}

	public int getMinPlayers() {
		return minPlayers;
	}

	public void setMinPlayers(int minPlayers) {
		this.minPlayers = minPlayers;
	}

	public int getMaxPlayers() {
		return maxPlayers;
	}

	public void setMaxPlayers(int maxPlayers) {
		this.maxPlayers = maxPlayers;
	}

	public List<GameTeam> getTeams() {
		return teams;
	}

	public void addTeam(GameTeam team) {
		team.setGame(this);
		teams.add(team);
	}

	public List<Match> getMatches() {
		return matches;
	}

	public void addMatch(Match match) {
		matches.add(match);
	}
}

File: application/app/src/main/java/com/yahyeet/boardbook/model/entity/Game.kt
package com.yahyeet.boardbook.model.entity

import java.util.ArrayList


class Game : AbstractEntity {
    var name: String? = null
    var description: String? = null
    var minPlayers: Int = 0
    var maxPlayers: Int = 0
    var difficulty: Int = 0
    private val teams = ArrayList<GameTeam>()
    private val matches = ArrayList<Match>()

    constructor(name: String,
                description: String,
                difficulty: Int,
                minPlayers: Int,
                maxPlayers: Int) {
        this.name = name
        this.description = description
        this.difficulty = difficulty
        this.minPlayers = minPlayers
        this.maxPlayers = maxPlayers
    }

    constructor(id: String) : super(id) {}


    //Nox has added this in purely for testing purposes
    constructor(name: String, description: String) {
        this.name = name
        this.description = description
    }

    fun getTeams(): List<GameTeam> {
        return teams
    }

    fun addTeam(team: GameTeam) {
        team.game = this
        teams.add(team)
    }

    fun getMatches(): List<Match> {
        return matches
    }

    fun addMatch(match: Match) {
        matches.add(match)
    }
}


